dotnet/src/SemanticKernel.Core/TemplateEngine/Blocks/CodeBlock.cs

```diff
@@ -170,12 +172,29 @@ private KernelArguments EnrichFunctionArguments(KernelArguments arguments)
             this.Logger.LogTrace("Passing variable/value: `{Content}`", firstArg.Content);
         }
 
+        string? positionalParameterName = null;
+        object? positionalInputValue = null;
         var namedArgsStartIndex = 1;
+
+        // Get the function metadata
+        var functionMetadata = kernel.Plugins.GetFunction(fBlock.PluginName, fBlock.FunctionName).Metadata;
+
+        // Check if the function has parameters to be set
+        if (functionMetadata.Parameters.Count == 0)
+        {
+            throw new ArgumentException($"Function {fBlock.PluginName}.{fBlock.FunctionName} does not take any arguments but it is being called in the template with {this._tokens.Count - 1} arguments.");
+        }
+
         if (firstArg.Type is not BlockTypes.NamedArg)
         {
-            object? input = ((ITextRendering)this._tokens[1]).Render(arguments);
+            // Gets the function first parameter name
+            positionalParameterName = functionMetadata.Parameters[0].Name;
+
+            positionalInputValue = ((ITextRendering)this._tokens[1]).Render(arguments);
+            // Type check is avoided and marshalling is done by the function itself
+
             // Keep previous trust information when updating the input
-            arguments[KernelArguments.InputParameterName] = input;
+            arguments[positionalParameterName] = positionalInputValue;
```

Comment:
nit: firstParameterName

Reply:
renamed to `firstPositionalParameterName`